idx/memory/memory.go

```diff
@@ -638,16 +766,16 @@ func (m *UnpartitionedMemoryIdx) add(archive *idx.Archive) {
 
 	statMetricsActive.Inc()
 
-	def := &archive.MetricDefinition
+	def := archive.MetricDefinitionInterned
 	path := def.NameWithTags()
 
 	if TagSupport {
 		// Even if there are no tags, index at least "name". It's important to use the definition
 		// in the archive pointer that we add to defById, because the pointers must reference the
 		// same underlying object in m.defById and m.defByTagSet
-		m.indexTags(def)
+		m.indexTags(archive.MetricDefinitionInterned)
```

Comment:
Can we just pass `def` from line 769? Or will this mess up the `release` process?